let promise = new Promise(function(resolve, reject) {
    // исполнитель (executor)
    // Эта функция-исполнитель запускается автоматически.

    // resolve/reject - колбэки, вызываются внутри исполнителя
    // исполнитель может вызвать только один из них. Елси вызовет оба - выполнится только
    // первый, следующий - будет проигнорирован

    
});

У объекта promise есть свойство [[PromiseState]] и свойство [[PromiseResult]].
Они доступны только через методы .then/.catch/.finally .

-- .then - принимает функцию, в аргументе которой результат УСПЕШНОГО промиса, далее
    выполняет с ним какую-то логику

-- .catch - принимает функцию, в аргументе которой результат НЕУСПЕШНОГО(с ошибкой) промиса.
    Далее обрабатывает ошибку

-- .finally - вне зависимости от успеха выполнения промиса, выполняет какую-то логику.
    Прокидывает дальше по цепочке промисов ошибку или успешный результат.